src/os_win/os_errno.c

```diff
@@ -63,14 +63,33 @@ int
 __wt_errno(void)
 {
 	/*
+	 * Check for 0:
+	 * It's easy to introduce a problem by calling the wrong error function,
+	 * for example, this function when the MSVC function set the C runtime
+	 * error value. Handle gracefully and always return an error.
+	 */
+	return (errno == 0 ? WT_ERROR : errno);
```

Comment:
@keithbostic This change to Windows error handling bothers me. The semantic for Windows calls is to check for failure via checking the return, and then call ``GetLastError`` once or check ``errno``.

I can see cases where our code doesn't adhere to that policy (e.g: truncate), but I think we should fix the code rather than resort to returning a generic WT_ERROR. Did you make this change to solve a particular problem?